packages/env/lib/config/config.js

```diff
@@ -273,15 +276,31 @@ function withOverrides( config ) {
 	// Override WordPress core with environment variable.
 	if ( process.env.WP_ENV_CORE ) {
 		const coreSource = includeTestsPath(
-			parseSourceString( process.env.WP_ENV_CORE, {
-				workDirectoryPath: config.workDirectoryPath,
-			} ),
-			{ workDirectoryPath: config.workDirectoryPath }
+			parseSourceString( process.env.WP_ENV_CORE, { workDirectoryPath } ),
+			{ workDirectoryPath }
 		);
 		config.env.development.coreSource = coreSource;
 		config.env.tests.coreSource = coreSource;
 	}
 
+	// If one of the sources is STILL null after parsing every config and CLI option,
+	// set it to the latest WordPress version. This also allows one to use
+	// .wp-env.override.json to reset the coreSource to the latest version.
+	if (
+		config.env.development.coreSource === null ||
```

Comment:
After thinking about this some more, I don't actually think we need this fallback. `WP_ENV_CORE` _can_ be set to `"null"` but that will be parsed as a download source instead of being recognized as `null`.

Running `WP_ENV_CORE=null npm run wp-env start` will throw an error.

Reply:
To clarify, do you mean that we need to also fix that case above or somehow integrate it with this fallback?

Reply:
No, I don't think handling that case is necessary. I more mean that it removes the need for the fallback altogether. The `WP_ENV_CORE` _requires_ a valid `core` configuration that _isn't_ `null` since that would be a string `"null"` instead of an actual `null`.

I propose the following changes to this PR:

* Remove `coreRef` lookup from `readConfig()` and use the default of `null` as it was before. Also remove `getCoreRef()`.
* Remove fallback from `withOverrides()` and return to previous state since `WP_ENV_CORE` can't be anything other than a valid ref.
* In `parseConfig()` add handling to replace a `null` `core` with the latest version. This feels like it makes the most sense because it's "parsing" the `null` into the appropriate version.

Reply:
That makes sense to me 👍 

I do wonder if it makes sense to allow some `WP_ENV_CORE` value which "unsets" whatever version would otherwise be used -- but also seems unimportant now.